### DISCUSSION TITLE
Using pre-built images with wave?

### DISCUSSION THREAD

**Discussion by stkgo** [nextflow-io/nextflow#5404] [Q&A]
Hello, I see that using fusion requires the use of wave. I already have a large collection of Docker images for my pipelines that I am happy with and do not want to rebuild with wave. Is there a way to use fusion and wave without being forced to rebuild all of my existing images with wave? I do see that registry "mirroring" is an option, however it's unclear to me if that will cause Nextflow to pull images from that registry or if it will only push to it.

**Comment by pditommaso** [comment]
the primary use case of Wave is to enable the use of Fusion \*without* rebuild your images. Have a look [here](https://www.nextflow.io/docs/latest/fusion.html#getting-started). Happy to assist for any doubt

**Reply by stkgo** [reply]
Just to clarify, does Wave still download the container specified by the `container` keyword in a process just as Nextflow already does without Wave and Fusion? I'm just a bit confused since when I read up on Wave it sounds like a key component of it is that it builds containers.

**Reply by pditommaso** [reply]
What wave does is to include the Fusion binary in the image by adding a new layer in the container manifest while is pulled the docker client (not by wave). Therefore there no build (unless you request it via the *freeze* option)
